pwiz_tools/Skyline/FileUI/PeptideSearch/ImportPeptideSearchDlg.cs

```diff
@@ -895,6 +895,25 @@ public void CloseWizard(DialogResult result)
             btnNext.Enabled = e.NumFoundFiles > 0;
         }
 
+        //
+        // Handler for Full Scan settings changes that require ion type changes
+        //
+        private void OnIonTypesChanged(object sender, FullScanSettingsControl.IonTypesChangedEventArgs e)
+        {
+            if (e.PeptidePrecursorIonTypeEnabled.HasValue && 
+                TransitionSettingsControl.PeptideIonTypes.Contains(IonType.precursor) != e.PeptidePrecursorIonTypeEnabled) // Full-Scan settings adjusted ion types to include or exclude "p"
+            {
+                var list = TransitionSettingsControl.PeptideIonTypes.ToList();
+                if (e.PeptidePrecursorIonTypeEnabled.Value)
+                    list.Add(IonType.precursor);
+                else
+                    list.Remove(IonType.precursor);
+                if (list.Count > 0)
+                    TransitionSettingsControl.PeptideIonTypes = list.ToArray();
+            }
+            // TODO(bspratt): handle small mol ion types when this gets extended for UIModes

```

Comment:
Can we make this a CONSIDER or FUTURE? Ideally, TODO should be for actionable work, and not future things we need to handle when something else changed.